### DISCUSSION TITLE
T-Pot preview dies and throws errors at restart - /opt/tpot/etc/tpot.yml: No such file or directory - Unit tpot.service could not be found.

### DISCUSSION THREAD

**Discussion by HachimanSec** [telekom-security/tpotce#1413] [Q&A]
Hi everyone, 

I deployed the preview version which was working fine until recently. Then the server started to die after about 24 hours. A reboot fixed the problem. 

Since today I run into this:
```
sudo /home/debian/tpotce/bin/dps.sh
cat: /opt/tpot/etc/tpot.yml: No such file or directory
[ ========| System |======== ]
     DATE:  Sat Sep 23 14:58:35 UTC 2023
   UPTIME:   14:58:35 up 0 min,  1 user,  load average: 8.57, 1.93, 0.64
Unit tpot.service could not be found.
    T-POT:  INACTIVE
BLACKHOLE:  DISABLED
```

tpot.yml is missing, tpot.service can't be found and t-POT is marked inactive. 

When I navigate to the map or the Kibana dashboard, I do see some activity. 

/data has enough space and is on a seperate partition. 

```
lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 11 (bullseye)
Release:	11
Codename:	bullseye

df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            7.2G     0  7.2G   0% /dev
tmpfs           1.5G  3.4M  1.5G   1% /run
/dev/sdb1        50G  9.0G   39G  20% /
tmpfs           7.2G     0  7.2G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sdb15      124M   11M  114M   9% /boot/efi
/dev/sda        127G   17G  105G  14% /data
tmpfs           1.5G     0  1.5G   0% /run/user/1000
```

Anyone any idea what could be wrong?

EDIT 2: Unfortunately the server died again. This is the last thing from the cloud console. 
Any idea which log to check?
<img width="724" alt="Screenshot 2023-09-24 at 17 36 39" src="https://github.com/telekom-security/tpotce/assets/20123996/90f17c29-4eae-40b1-a020-9df530f1c3f2">

EDIT 1: Added a docker restart in the crontab, so far it works fine. Lets see the next days. 
```
@reboot docker compose -f /<path_to_tpot_>/tpotce/preview/docker-compose.yml down -v; \
docker container prune -f; \
docker image prune -f; \
docker compose -f /<path_to_tpot_>/tpotce/preview/docker-compose.yml up -d
```

Thx!

**Comment by t3chn0m4g3** [comment]
Please follow the Preview Documentation [here](https://github.com/telekom-security/tpotce/tree/master/preview). The concept of the preview is quite different from ISO or Post Install based installations, i.e. there is no `tpot.yml` and everything necessary for starting T-Pot can be found in the `preview` folder.

> @reboot docker compose -f /<path_to_tpot_>/tpotce/preview/docker-compose.yml down -v; \
docker container prune -f; \
docker image prune -f; \
docker compose -f /<path_to_tpot_>/tpotce/preview/docker-compose.yml up -d

The command is incorrect, you need to adjust to the correct path.

**Reply by HachimanSec** [reply]
Thx for the link. This is which I used to instal the preview. sorry for being unclear about it. 
So I assume that the notifications can be ignored and are a leftover? 

Do you have any idea why the preview would regularly hang? It seems like one day is the max I can get it up and running. 

PS: The crontab paths have been modified in my config. I just copy and paste the default from the doc.

PPS: I got most request on the DDoSpot. Could this be the source of the problem? 
Just right now I can access the map but for example the Kibana dashboard at port 64297 became unresponsive. 
Might it be because of the RAM being on the lower end? 

<img width="1785" alt="Screenshot 2023-09-25 at 13 52 49" src="https://github.com/telekom-security/tpotce/assets/20123996/c12e2fb5-19cc-4f7b-8956-061663c4d7d6">

**Reply by t3chn0m4g3** [reply]
You can remove `ddospot` from the `docker-compose.yml` to reduce the load on your machine.
The reason why the system hangs may be specific to your cloud provider, mine runs stable for a couple of months now.

**Reply by HachimanSec** [reply]
Running the crontab entry twice a day seems to help. Brazil is quite "active" on this one. 
Removing DDoS might be a workaround if it gets worse. Thx @t3chn0m4g3 👍🏼
<img width="1726" alt="Screenshot 2023-09-25 at 19 11 57" src="https://github.com/telekom-security/tpotce/assets/20123996/792b2e1a-7068-42af-8c97-4ce55f5fa34a">

**Reply by HachimanSec** [reply]
I think I found something in the logs. Ciscoasa and adbhoney seem to be the ones causing the halt of the system through aleged SYN flooding. 

Anyone ever experienced this? Any idea what could be done - other then simply disabling? 
![Screenshot 2023-09-26 at 17 08 52](https://github.com/telekom-security/tpotce/assets/20123996/4b4f0b32-7b04-4bde-9312-617b2e65614f)
